### PR TITLE
Refactor code for Extension API doc (#407)

### PR DESCRIPTION
**Pull request by FabienLelaquais**

### CODE REVIEW

File: src/taipy/gui/extension/library.py [hunk 1]
```diff
@@ -22,74 +22,78 @@
...
 
 
 class Element:
     """
-    TODO
-    This is instantiated to describe an element.
+    The definition of a custom visual element.
 
+    The definition of an element is made of its names, its properties, and
+    the
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/extension/library.py]: Unfinished business?

File: src/taipy/gui/extension/library.py [hunk 2]
```diff
@@ -162,52 +166,71 @@ def render(self, gui: "Gui", properties: t.Dict[str, t.Any], hash_names: t.Dict[
...
         """
         if hasattr(self, "_render") and callable(self._render):
             self._render(gui, properties, hash_names, builder)
 
 
 class ElementLibrary(ABC):
     """
-    TODO
-    This class needs to be inherited to define a new library.
+    An library of user-defined visual elements.
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/extension/library.py]: A

File: src/taipy/gui/extension/library.py [hunk 3]
```diff
@@ -162,52 +166,71 @@ def render(self, gui: "Gui", properties: t.Dict[str, t.Any], hash_names: t.Dict[
...
         """
         return NotImplemented
 
+    def get_js_module_name(self) -> str:
+        """
+        Returns the js module name (default to self.get_name()).
+        Must be unique on the browser window scope.
+
+        """
+        return self.get_name()
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/extension/library.py]: Use Carmel case?

**Review by FredLL-Avaiga — APPROVED** [review]
A few comments

File: src/taipy/gui/extension/library.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/extension/library.py]: Indeed. I was waiting for your own changes before adding anything can could turn wrong.

File: src/taipy/gui/extension/library.py (hunk 2 above)
Comment (FabienLelaquais) [reply src/taipy/gui/extension/library.py]: ✔️

File: src/taipy/gui/extension/library.py (hunk 3 above)
Comment (FabienLelaquais) [reply src/taipy/gui/extension/library.py]: Indeed.

File: src/taipy/gui/extension/library.py [hunk 4]
```diff
@@ -34,94 +34,93 @@ class ElementProperty:
...
+        return self._js_name or _to_camel_case(name)
 
 class Element:
     """
     The definition of a custom visual element.
 
-    The definition of an element is made of its names, its properties, and
-    the
-    TODO
+    An element is defined by its properties (name, type and default value) and
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/extension/library.py:75]: name ?

File: src/taipy/gui/gui.py [hunk 5]
```diff
@@ -271,8 +271,12 @@ def __init__(
...
     @staticmethod
     def add_library(library: ElementLibrary):
-        """
-        TODO
+        """Add a custom visual element library to this Gui instance.
+
+        Arguments:
+            library: The custom visual element library to add to this library.
+
+        TODO: What if we add two libraries with the same name?
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py:279]: done

**Review by FredLL-Avaiga — APPROVED** [review]
let's do this

File: src/taipy/gui/extension/library.py (hunk 4 above)
Comment (FabienLelaquais) [reply src/taipy/gui/extension/library.py:75]: Correct. No name!

Comment (FabienLelaquais) [reply src/taipy/gui/extension/library.py:75]: Yes name! The name... of the properties.

**Pull request merged**